projects/bootloader/test/test_config.c

```diff
@@ -0,0 +1,171 @@
+  persist_ctrl_periodic(&s_test_persist_storage_1, false);
+
+  persist_init(&s_test_persist_storage_2, BOOTLOADER_CONFIG_PAGE_2_FLASH_PAGE, &s_test_config_2,
+               sizeof(s_test_config_2), false);
+  persist_ctrl_periodic(&s_test_persist_storage_2, false);
+
+  TEST_ASSERT_EQUAL_MEMORY(&s_test_config_1, &s_test_config_2, sizeof(s_test_config_1));
+
+  // Test 3, testing if config_init throws the correct error if both pages are corrupted
+  s_test_config_1.controller_board_ID = 6;
+  s_test_config_1.app_code = 6;
+  s_test_config_1.app_size = 6;
+  persist_commit(&s_test_persist_storage_1);
+
+  s_test_config_2.controller_board_ID = 5;
+  s_test_config_2.app_code = 5;
+  s_test_config_2.app_size = 5;
+  persist_commit(&s_test_persist_storage_2);
+
+  TEST_ASSERT_EQUAL(STATUS_CODE_INTERNAL_ERROR, config_init());
+}
+
+void test_config_get(void) {
+  // This test tests to see that if config_get sets the given
+  // pointer to config page 1
+  persist_init(&s_test_persist_storage_1, BOOTLOADER_CONFIG_PAGE_1_FLASH_PAGE, &s_test_config_1,
+               sizeof(s_test_config_1), false);
+  persist_ctrl_periodic(&s_test_persist_storage_1, false);
+
+  BootloaderConfig test_input_config = { 0 };
+  TEST_ASSERT_OK(config_get(&test_input_config));
+
+  TEST_ASSERT_EQUAL_MEMORY(&s_test_config_1, &test_input_config, sizeof(s_test_config_1));
+}
+
+void test_config_commit(void) {
+  // I need to create an input config to give config_commit();
+  // I use random values for the input_config
+  BootloaderConfig test_input_config = { .CRC32 = 1,
+                                         .controller_board_ID = 1,
+                                         .controller_board_name = { 'a', '\0' },
+                                         .project_present = true,
+                                         .project_name = { 'a', '\0' },
+                                         .project_info = { 'a', '\0' },
+                                         .git_version = { 'a', '\0' },
+                                         .app_code = 1,
+                                         .app_size = 1 };
+
+  // Test 1, testing if page 2 and page 1 are both the new page
+  config_commit(&test_input_config);
+
+  persist_init(&s_test_persist_storage_1, BOOTLOADER_CONFIG_PAGE_1_FLASH_PAGE, &s_test_config_1,
+               sizeof(s_test_config_1), false);
+  persist_ctrl_periodic(&s_test_persist_storage_1, false);
+
+  persist_init(&s_test_persist_storage_2, BOOTLOADER_CONFIG_PAGE_2_FLASH_PAGE, &s_test_config_2,
+               sizeof(s_test_config_2), false);
+  persist_ctrl_periodic(&s_test_persist_storage_2, false);
+
+  TEST_ASSERT_EQUAL_MEMORY(&s_test_config_1, &test_input_config, sizeof(s_test_config_1));
+  TEST_ASSERT_EQUAL_MEMORY(&s_test_config_2, &s_test_config_1, sizeof(s_test_config_2));
+
+  // Test 2, testing if page 1 is reset when the input config was corrupted
+  s_function_pointer_hook = &prv_change_config_page;
```

Comment:
You don't need the `&` (and we don't use it in the rest of the codebase), just setting it to the function itself will auto convert to a function pointer.